src/main/java/formflow/library/ScreenController.java

```diff
@@ -131,13 +131,13 @@ ModelAndView postScreen(
     // if there's already a session
     if (submission.getId() != null) {
       submission.mergeFormDataWithSubmissionData(formSubmission);
-      saveToRepository(submission);
     } else {
       submission.setFlow(flow);
       submission.setInputData(formSubmission.getFormData());
-      saveToRepository(submission);
-      httpSession.setAttribute("id", submission.getId());
     }
+    handleBeforeSaveAction(getScreenConfig(flow, screen), submission);
+    saveToRepository(submission);
+    httpSession.setAttribute("id", submission.getId());
```

Comment:
This is probably already set most of the time, but it seems ok to set it again anyways?

Reply:
Yeah, I don't think this will be an issue.